ci/Jenkinsfile-pyp

```diff
@@ -0,0 +1,205 @@
+            }
+            steps {
+                dir('inmanta-sphinx') {
+                    withCredentials([string(credentialsId: 'fff7ef7e-cb20-4fb2-a93b-c5139463c6bf', variable: 'GITHUB_TOKEN')]) {
+                        sh '''
+                            git checkout next
+
+                            # Cherry-picking the patch commits on the next branch and setting
+                            # the changelog correctly is done manually before this job runs.
+
+                            export GIT_ASKPASS="${WORKSPACE}/ASK_PASS"
+                            ${WORKSPACE}/env/bin/python3 -m pip install -U pip bumpversion
+
+                            # Bump version
+                            ${WORKSPACE}/env/bin/bumpversion --commit ${version_increment}
+                            git push origin next
+                        '''
+                    }
+                }
+            }
+        }
+
+        stage("Add tag") {
+            when {
+                allOf {
+                    expression { "${run_job}" == "true" }
+                    expression { "${release_type}" == "stable" }
+                }
+            }
+            steps{
+                dir('inmanta-sphinx') {
+                    withCredentials([string(credentialsId: 'fff7ef7e-cb20-4fb2-a93b-c5139463c6bf', variable: 'GITHUB_TOKEN')]) {
+                        sh '''
+                            export GIT_ASKPASS="${WORKSPACE}/ASK_PASS"
+                            current_version=$(sed -n -E 's/current_version[[:space:]]*=[[:space:]]*([\\digit.]*)/\\1/p' .bumpversion.cfg)
+                            tag="v${current_version}"
+                            git tag -a -m "Release: ${current_version}" ${tag}
+                            git push origin ${tag}
+                        '''
+                    }
+                }
+            }
+        }
+
+        stage("Build dist") {
+            when {
+                expression { "${run_job}" == "true" }
+                anyOf {
+                    expression { "${release_type}" == "stable" }
+                    expression { "${release_type}" == "dev" }
+                }
+            }
+            steps {
+                dir('inmanta-sphinx') {
+                    sh '''
+                        ${WORKSPACE}/env/bin/python3 -m pip install -U twine pip bumpversion
+                        if [ "${release_type}" == "stable" ]; then
+                            ${WORKSPACE}/env/bin/python3 setup.py egg_info -Db "" sdist
+                        else
+                            ${WORKSPACE}/env/bin/python3 setup.py sdist
+                        fi
+                    '''
+                }
+            }
+        }
+
+        stage("Publish to PyPi") {
+            when {
+                allOf {
+                    expression { "${run_job}" == "true" }
+                    expression { "${upload}" == "true" }
+                    expression { "${release_type}" == "stable" }
```

Comment:
The previous build step allows building a dev release, but this step doesn't allow you to upload it to PyPi.